src/main/java/yarmark/paint/PencilTool.java

```diff
@@ -27,8 +29,9 @@ public void mouseReleased(Graphics g, int x, int y) {
 	}
 
 	@Override
-	public void mouseDragged(Graphics g, int x, int y) {
+	public void mouseDragged(Graphics2D g, int x, int y) {
 		g.setColor(properties.getColor());
+		g.setStroke(new BasicStroke(properties.getWeight()));
```

Comment:
Don't create a new BasicStroke object every single time the mouse is pressed. Move BasicStroke into PaintProperties and do this for mousePressed also